runtime/src/accounts_db.rs

```diff
@@ -4430,14 +4457,26 @@ impl AccountsDb {
         slot: Slot,
         ancestors: &Ancestors,
         check_hash: bool,
+        can_cached_slot_be_unflushed: bool,
     ) -> Result<(Hash, u64), BankHashVerificationError> {
         if !use_index {
+            let accounts_cache_and_ancestors = if can_cached_slot_be_unflushed {
+                Some((&self.accounts_cache, ancestors, &self.accounts_index))
+            } else {
+                None
+            };
+
             let mut collect_time = Measure::start("collect");
             let (combined_maps, slots) = self.get_snapshot_storages(slot, Some(ancestors));
             collect_time.stop();
 
             let mut sort_time = Measure::start("sort_storages");
-            let storages = SortedStorages::new_with_slots(combined_maps.iter().zip(slots.iter()));
+            let min_root = self.accounts_index.min_root();
+            let storages = SortedStorages::new_with_slots(
+                combined_maps.iter().zip(slots.iter()),
+                min_root,
```

Comment:
Not a big deal, but one thing to note is there can potentially be a large gap between the min root and the first slot in the cache (we keep around 425,000 roots, but we only keep around the latest 10 seconds of roots at startup in the cache). If we ever need to repeatedly call this function, might be more efficient to either:

1) Have a separate min for the cache
2) Cache the starting point for the cache search after performing it the first time.